app/src/main/java/ingsw/group1/findmyphone/LocationManager.java

```diff
@@ -211,11 +210,12 @@ public void onFailure(@NonNull Exception e) {
      * @param mapsLatitude latitude extracted by response sms
      * @param mapsLongitude longitude extracted by response sms
      */
-    public void OpenMapsUrl(Double mapsLatitude, Double mapsLongitude)
+    public void OpenMapsUrl(Context context, Double mapsLatitude, Double mapsLongitude)
     {
         String url = MAPS_START_URL + mapsLatitude + "," + mapsLongitude;
         Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
-        currentContext.getApplicationContext().startActivity(intent);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        context.getApplicationContext().startActivity(intent);
```

Comment:
Well yes this will solve it 99%. The Context was never assigned so obviously it would not work

Reply:
As far as I can tell this solves the failing of the location request, I saw that both requests and responses were correctly handled, it was just the map that was failing to open